### PR TITLE
remove sky storage dependency

### PR DESCRIPTION
**Pull request by jlewitt1**

### CODE REVIEW

**Comment by jlewitt1** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/run-house/runhouse/415?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #414** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/414?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #415** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/415?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/415?utm_source=stack-comment).

**Comment by jlewitt1** [comment]
@carolineechen for now we are using subprocess commands (which is what sky was doing under the hood), maybe longer term we move to using boto or gsutil directly but thinking for now this should do for removing ouright all the sky dependencies. wdyt?

File: runhouse/resources/folders/s3_folder.py [hunk 1]
```diff
@@ -61,18 +57,20 @@ def _download(self, dest):
...
+                stderr=subprocess.DEVNULL,
+                check=True,
+            )
+        except subprocess.CalledProcessError as e:
+            raise e
 
-        download_command = S3CloudStorage().make_sync_dir_command(src, dest)
-        return download_command
+    def _download_command(self, src: str, dest: str):
+        # https://github.com/skypilot-org/skypilot/blob/983f5fa3197fe7c4b5a28be240f7b027f7192b15/sky/cloud_stores.py#L133
```
Comment (carolineechen) [inline runhouse/resources/folders/s3_folder.py:71]: should this be https://github.com/skypilot-org/skypilot/blob/983f5fa3197fe7c4b5a28be240f7b027f7192b15/sky/cloud_stores.py#L68?

File: runhouse/resources/folders/s3_folder.py [hunk 2]
```diff
@@ -124,3 +117,130 @@ def _to_data_store(
...
+        if timeout:
+            logger.info(
+                f"Transfer timed out after {(time.time() - start) / TIMEOUT_SECONDS:.2f} "
+                "hours. Please check the status of the transfer job in the GCP "
+                "Storage Transfer Service console at "
+                "https://cloud.google.com/storage-transfer-service"
+            )
+
+    @staticmethod
+    def add_bucket_iam_member(
```
Comment (carolineechen) [inline runhouse/resources/folders/s3_folder.py:232]: is this relevant to s3 or just google cloud?

**Review by carolineechen — COMMENTED** [review]
yea this approach sounds good for now and overall code looks good, thanks! have these updates been tested?

File: runhouse/resources/folders/s3_folder.py (hunk 2 above)
Comment (jlewitt1) [reply runhouse/resources/folders/s3_folder.py:232]: oh good point its just relevant for GCS, will move into the google folder's class

**Comment by jlewitt1** [comment]
these changes were tested, although the folder tests are still in flight so that will need to continue to be fleshed out.. but feels out of scope for this PR

**Review by carolineechen — APPROVED** [review]
think this should be good to go once it's rebased on main -- ran it on the updated folder tests and behavior is on par with the main branch
